### PR TITLE
zynq7000-flash: timeouts rework

### PR DESCRIPTION
**Pull request by badochov**
JIRA: RTOS-481

## Description
Changed timeouts to prevent timeouts on starvation.

## Motivation and Context
https://github.com/phoenix-rtos/phoenix-rtos-project/issues/725

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: zynq-7000.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/zynq7000-flash/qspi.c
Comment (agkaminski) [inline storage/zynq7000-flash/qspi.c:44]: clang-format is wrong (it thinks it's a function), restore previous version

File: storage/zynq7000-flash/qspi.c
Comment (agkaminski) [inline storage/zynq7000-flash/qspi.c:209]: For line breaks with a body use (if, for, while) use:
- double indentation for a condition part (only tabs, no spaces),
- single indentation for a body.

**Review by agkaminski — COMMENTED** [review]
Logic LGTM, small style remarks

**Review by agkaminski — APPROVED** [review]
LGTM

**Comment by badochov** [comment]
Let's wait for @maska989 to confirm before merge.

File: storage/zynq7000-flash/qspi.c
Comment (kemonats) [inline storage/zynq7000-flash/qspi.c:215]: Faster, without divide 64 bit numbers.
```suggestion
		time_t now;
		time_t end;
		gettime(&now, NULL);
		end = now + timeout * 1000;
		while ((now < end) && ((*(qspi_common.base + sr) & (1 << 2)) == 0)) {
			err = condWait(qspi_common.cond, qspi_common.irqLock, end - now);
			gettime(&now, NULL);
		}
```

**Comment by maska989** [comment]
Tested on the machine, test campaign done about 150 times on two targets without jffs2 error or other hiccups
LGTM and thanks 4 patience

File: storage/zynq7000-flash/qspi.c
Comment (agkaminski) [reply storage/zynq7000-flash/qspi.c:215]: I would be best to remove `qspi_timeMsGet()` altogether then

**Review by maska989 — APPROVED** [review]
LGTM

**Review by agkaminski — APPROVED** [review]

**Pull request merged**
